src/App.js

```diff
@@ -1,26 +1,127 @@
+    e.preventDefault();
+    this.fetchWords(this.state.searchWords)
+  }
+
+  searchDatesSubmit = (e) => {
+    e.preventDefault();
+    this.fetchDate(this.state.searchDates)
+  }
+
+  searchAuthorSubmit = (e) => {
+    e.preventDefault();
+    this.fetchAuthor(this.state.searchAuthor)
+  }
+
+  fetchWords = (search) => {
+    fetch (`http://hn.algolia.com/api/v1/search?query=${search}&tags=story`)
+    .then(response => response.json())
+    .then(data => {
+      if (data.hits.length > 0) {
+        this.setState({stories: data.hits})
+      } else {
+        this.setState("No results found. Please search again.")
+      }
+    })
+    .catch(error => console.log("Parsing failed: ", error))
+  }
+
+  fetchDate = (search) => {
+    fetch (`http://hn.algolia.com/api/v1/search_by_date?query=${search}&tags=story`)
+    .then(response => response.json())
+    .then(data => {
+      if (data.hits.length > 0) {
+        this.setState({stories: data.hits})
+      } else {
+        this.setState("No results found. Please search again.")
+      }
+    })
+    .catch(error => console.log("Parsing failed: ", error))
+  }
+
+  fetchAuthor = (search) => {
+    fetch (`http://hn.algolia.com/api/v1/search?tags=author_${search}&tags=story`)
+    .then(response => response.json())
+    .then(data => {
+      if (data.hits.length > 0) {
+        this.setState({stories: data.hits})
+      } else {
+        this.setState("No results found. Please search again.")
+      }
+    })
+    .catch(error => console.log("Parsing failed: ", error))
+  }
+
+  render() {
+    return (
+      <div className="page-container">
+        <div className="form-container">
+          <form>
+            <input
+              name='searchWords'
+              placeholder="Search story titles"
+              value={this.state.searchWords}
+              onChange={this.onChange}
+            ></input>
+            <button onClick={this.searchWordsSubmit}>Search</button>
+          </form>
+          <br/>
+          <form>
+            <input
+              name='searchDates'
+              type='date'
+              placeholder="Search story dates"
+              value={this.state.searchDates}
+              onChange={this.onChange}
+            ></input>
+            <button onClick={this.searchDatesSubmit}>Search</button>
+          </form>
+          <br/>
+          <form>
+            <input
+              name='searchAuthor'
+              placeholder="Search story authors"
+              value={this.state.searchAuthor}
+              onChange={this.onChange}
+            ></input>
+            <button onClick={this.searchAuthorSubmit}>Search</button>
+          </form>
+          {/* <h2>You searched: {this.state.search}</h2> */}
+        </div>
+        <div className="results-container">
+          {/* {console.log(`is this right: `, this.state.stories)} */}
+          {this.state.stories.map ((stories, i) => (
+            <div>
+              <div key={i+1}>
```

Comment:
it's ok for key to be `0` so no need to add 1 here.  IRL you usually want it to be something unique (like an id) esp if the order of the items can change (since React uses the key to determine what it needs to rerender)